### DISCUSSION TITLE
Filter/hook on each select queries

### DISCUSSION THREAD

**Discussion by or-deal** [SeaQL/sea-orm#2012] [Q&A]
Hello,
I'm working on an API with a rate limit, so I have a table that distributes tokens like :
```sql
/* TODO : add a uniqueness constraint to the owner_id and token combination */
CREATE TABLE `token` (
    `id`          BIGINT UNSIGNED PRIMARY KEY NOT NULL AUTO_INCREMENT,
    `owner_id`    BIGINT UNSIGNED NOT NULL, -- app id
    `token`       VARCHAR(36)  NOT NULL, -- uuid v4
    `usage`       BIGINT UNSIGNED NOT NULL, -- max 500 for unvalidated app (dev purpose), 10 000 for validated app
    `last_update` TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
);
```
What I'd like to do, is for each select automatically call a filter (or hook) that looks at whether `last_update` is today, and if it's not today, resets to zero `usage`, updates and returns the updated entity.

Something like : 
```rust
impl SomethingForToken {
    pub async fn on_select(&self, token) -> TokenModel {
        // some token.last_update check
        if need_reset {
            let mut tkn = token.unwrap().into();
            tkn.usage = Set(0);
            return tkn.update(self.db).await;
        }
        token
    }
}
```
Currently, it doesn't seem to exist, so i wrap all my select in a function (`real_usage_hook(db, token) -> Result`)... On a small application this isn't a problem, but on a larger one it can quickly become a source of forgetfulness.

**Comment by tyt2y3** [comment]
Interesting. the `after_select` idea has been discussed in #1991, I don't think it is a good idea to make it async though. It'd probably be better to send a message via a channel and let another thread handle the reset.
